index.js

```diff
@@ -4,22 +4,15 @@ import {lineIterator, combineAsyncIterables} from './utilities.js';
 export default function nanoSpawn(command, arguments_ = [], {signal, timeout, nativeOptions} = {}) {
 	const subprocess = spawn(command, arguments_, {...nativeOptions, signal, timeout});
 
-	// eslint-disable-next-line no-async-promise-executor
-	const promise = new Promise(async (resolve, reject) => {
-		try {
-			subprocess.on('close', exitCode => {
-				// TODO: Pass in `stdin` and `stdout` strings here.
-				resolve({
-					exitCode,
-				});
-			});
+	const promise = new Promise((resolve, reject) => {
+		subprocess.on('close', (exitCode, signalName) => {
```

Comment:
I double checked the core Node.js code, and just listening for the `close` and `error` events catches every possible exit situation. :+1: 

Also, unlike the `exit` event, the `close` event waits for IPC and file descriptors to close.